q2_feature_classifier/_cutter.py

```diff
@@ -109,17 +108,22 @@ def _gen_reads(sequences: DNAIterator,  f_primer: str, r_primer: str,
         if not amp:
             amp = _approx_match(seq, f_primer, r_primer, identity)
         if not amp:
-            amp = _approx_match(seq.reverse_complement(), f_primer, r_primer,
-                                identity)
+            amp = _approx_match(
+                seq.reverse_complement(), f_primer, r_primer, identity)
+        if not amp:
+            continue
+        if trunc_len > 0:
+            amp = amp[:trunc_len]
+        if trim_left > 0:
+            amp = amp[trim_left:]
         if not amp:
             continue
-        if length > 0:
-            amp = amp[:length]
         yield amp
 
 
 def extract_reads(sequences: DNAIterator,  f_primer: str, r_primer: str,
-                  length: int=-1, identity: float=0.8) -> DNAIterator:
+                  trunc_len: int=-1, trim_left: int=-1,
```

Comment:
Is `0` a useful value? I think it could make sense as the sentinel value. Otherwise this seems fine.

I think a perfect solution would be:
`trunc_len: Union[str, int]='None'` 
with the primitive type being:
 `Int % Range(0, None) | Str % Choices({'None'})` 
but we don't support that at the moment.